vllm/v1/engine/core.py

```diff
@@ -259,14 +265,37 @@ def run_busy_loop(self):
     def _handle_new_input(self):
         """Handle new input from the AsyncLLMEngine for async mode."""
 
+        def process_add_request(request_data: bytes) -> None:
+            engine_core_request: EngineCoreRequest = \
+                    self.msgpack_add_request_decoder.decode(request_data)
+            self.add_request(engine_core_request)
+
+        def process_abort_requests(request_data: bytes) -> None:
+            abort_requests: EngineCoreAbortRequest = \
+                    self.msgpack_abort_requests_decoder.decode(request_data)
+            self.scheduler.finish_requests(abort_requests.request_ids,
+                                           abort_requests.request_status)
+
         try:
             if self.input_socket.poll(timeout=0) != 0:
                 frames = self.input_socket.recv_multipart(copy=False)
-                engine_core_request = self.msgpack_decoder.decode(
+                # frames[0] identifies the type of the request and
+                # frames[1] is the actual request data
+                assert len(frames) == 2
+                request_type, request_data = frames[0], frames[1]
+
+                # Determine request_type
+                request_type = self.msgpack_request_type_decoder.decode(
                     frames[0].buffer)
```

Comment:
```suggestion
                request_type = self.msgpack_request_type_decoder.decode(
                    request_type.buffer)
```
except that per my other comment I don't think we need to decode at all here, the if below can just check e.g. `bytes(request_type.buffer) == ENGINE_CORE_ADD_REQ`.